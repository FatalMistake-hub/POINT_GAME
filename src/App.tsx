// App.tsx
import "./App.css";
import Container from "./component/Container";
import Header from "./component/Header";
import { useAppState } from "./hook/useAppState";

function App() {
  const [state, handlers] = useAppState();

  return (
    <div className="main">
      <Header {...state} {...handlers} />
      <Container {...state} {...handlers} />
    </div>
  );
}

export default App;