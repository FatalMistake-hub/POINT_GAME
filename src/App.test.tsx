import { render } from '@testing-library/react';
import { describe, it, expect } from 'vitest';
import App from './App'; // Adjust the import path as necessary

describe('App Component', () => {
  it('renders correctly and matches snapshot', () => {
    const { asFragment } = render(<App />);
    expect(asFragment()).toMatchSnapshot();
  });
});
