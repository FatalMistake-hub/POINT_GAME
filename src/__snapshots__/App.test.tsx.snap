// Vitest Snapshot v1

exports[`App Component > renders correctly and matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="main"
  >
    <div
      class="header"
    >
      <h2
        class="header__title info"
      >
        LET'S PLAY
      </h2>
      <div
        class="header__config"
      >
        <div
          class="header__config-label"
        >
          <span>
            Points:
          </span>
        </div>
        <div
          class="header__config-input"
        >
          <input
            max="1000"
            min="0"
            placeholder="Enter point"
            type="number"
            value="0"
          />
        </div>
      </div>
      <div
        class="header__time"
      >
        <div
          class="header__time-label"
        >
          <span>
            Time:
          </span>
        </div>
        <div
          class="header__time-value"
        >
          <span>
            0.0s
          </span>
        </div>
      </div>
      <div
        class="header__restart"
      >
        <button
          class="header__restart-button"
        >
          Play
        </button>
      </div>
    </div>
    <div
      class="container"
    />
  </div>
</DocumentFragment>
`;
